### ISSUE TITLE
dart fix adds 'flutter_gen: any' to pubspec.yaml which breaks app_localizations import reference

### ISSUE THREAD

**Issue by jamesncl** [flutter/flutter#148949]
### Steps to reproduce

I'm using Flutter 3.22.1. After running `dart fix --apply --code=use_super_parameters`, and then running `flutter pub get`, I found that my project could no longer find my generated `app_localizations.dart`:

> Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'

It seems that if you have an import to app_localizations in the same file as a target for the use_super_parameters fix, for example:

```
import 'package:flutter_gen/gen_l10n/app_localizations.dart';
import 'package:flutter/material.dart';

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final localisedText = AppLocalizations.of(context)!;
    return Text(localisedText.helloWorld);
  }
}

```

When you run `dart fix --apply --code=use_super_parameters`, it fixes the super parameter usage, but for some reason it also adds `flutter_gen: any` to your pubspec.yaml:

> dart fix --apply --code=use_super_parameters
> Computing fixes in fix_super_bug... 1.7s
> Applying fixes...                      0.0s
> 
> lib/main.dart
>   use_super_parameters • 1 fix
> 
> pubspec.yaml
>   MISSING_DEPENDENCY • 1 fix
> 
> 2 fixes made in 2 files.

```
dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter

  intl: any

  flutter_gen: any   # <--- This is added
```

Then when you next run `flutter pub get`, this causes all imports for `package:flutter_gen/gen_l10n/app_localizations.dart` to not be found.

To reproduce:

* Create a blank flutter project
Then follow [the instructions for internationalizing with custom strings](https://docs.flutter.dev/ui/accessibility-and-internationalization/internationalization):

* Add flutter_localizations and intl

```
flutter pub add flutter_localizations --sdk=flutter
flutter pub add intl:any
```
* Add generate: true to pubspec.yaml:
```
flutter:
  generate: true # Add this line
```
* Add l10n.yaml file to the root of project with:
```
arb-dir: lib/l10n
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
```
* Add `app_en.arb` to `${FLUTTER_PROJECT}/lib/l10n` with:
```arb
{
  "helloWorld": "Hello World!",
  "@helloWorld": {
    "description": "The conventional newborn programmer greeting"
  }
}
```

* Create a skeleton main.dart with a reference to the helloWorld string, and an old style usage of super parameter (see Code Sample below)

* Run `dart fix --apply --code=use_super_parameters`
* Run `flutter pub get`

### Expected results

The super parameter usage is updated to the new `super.key` style

### Actual results

The super parameter usage is updated to the new `super.key` style, but also `pubspec.yaml` is also modified so that it now shows as:

```
name: fix_super_bug
description: "A new Flutter project."
# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

version: 1.0.0+1

environment:
  sdk: '>=3.4.1 <4.0.0'

dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter

  intl: any

  flutter_gen: any   # <--- This has been added
  
dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0

flutter:
  generate: true
  uses-material-design: true

```

and `import 'package:flutter_gen/gen_l10n/app_localizations.dart';` is now an error:

> Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'

### Code sample

<details open><summary>Code sample</summary>

### main.dart

```dart
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
import 'package:flutter_gen/gen_l10n/app_localizations.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Localizations Sample App',
      localizationsDelegates: [
        GlobalMaterialLocalizations.delegate,
        GlobalWidgetsLocalizations.delegate,
        GlobalCupertinoLocalizations.delegate,
      ],
      supportedLocales: [
        Locale('en'), // English
        Locale('es'), // Spanish
      ],
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final localisedText = AppLocalizations.of(context)!;
    return Text(localisedText.helloWorld);
  }
}

```

### pubspec.yaml

```
name: fix_super_bug
description: "A new Flutter project."
# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

version: 1.0.0+1

environment:
  sdk: '>=3.4.1 <4.0.0'

dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter

  intl: any
  

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0

flutter:
  generate: true
  uses-material-design: true

```

### app_en.arb

```
{
  "helloWorld": "Hello World!",
  "@helloWorld": {
    "description": "The conventional newborn programmer greeting"
  }
}
```

### l10n.yaml

```
arb-dir: lib/l10n
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
untranslated-messages-file: untranslated.txt
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.1, on macOS 14.4.1 23E224 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✓] Android Studio (version 2023.3)
[✓] VS Code (version 1.89.1)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for Test user’s Apple Watch. Ensure the device is unlocked and discoverable via Bluetooth. (code -27)
    ! Error: Browsing on the local area network for Test user’s iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local area
      network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

! Doctor found issues in 1 category.

```

</details>

**Comment by jamesncl** [comment]
I just encountered the same thing when running `dart fix --apply --code=unnecessary_brace_in_string_interps`, so this seems to be a general problem with dart fix

**Comment by darshankawar** [comment]
Thanks for the very detailed report and replicable steps @jamesncl 
I followed the same and was able to replicate the reported behavior in stable, but following same steps in latest master doesn't give me the reported error. Below are details:

1. In stable, upon running `dart fix --apply --code=use_super_parameters`, it gives:

```
Computing fixes in local_issue_s... 6.6s
Applying fixes...                      0.0s

lib/main.dart
  use_super_parameters • 1 fix

pubspec.yaml
  MISSING_DEPENDENCY • 1 fix

2 fixes made in 2 files.
dhs@Dhss-MacBook-Pro local_issue_s % 
```

While the same in master gives:

```
dhs@Dhss-MacBook-Pro local_issue % dart fix --apply --code=use_super_parameters
Computing fixes in local_issue... 7.7s
Applying fixes...                      0.0s

lib/main.dart
  use_super_parameters • 1 fix

1 fix made in 1 file.
dhs@Dhss-MacBook-Pro local_issue % flutter pub get
Resolving dependencies... 
Downloading packages... 
  meta 1.14.0 (1.15.0 available)
Got dependencies!
1 package has newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
dhs@Dhss-MacBook-Pro local_issue % 
```

In master, the pubspec.yaml upon running above command doesn't add extra dependency:

<img width="405" alt="Screenshot 2024-05-24 at 12 10 42 PM" src="https://github.com/flutter/flutter/assets/67046386/7ed06c8e-3a3e-45bb-ace1-6f74768a2cf6">

You may confirm the same by switching to latest master.
Also see this issue for more reference: https://github.com/flutter/flutter/issues/148333

**Comment by jamesncl** [comment]
Thanks, so I guess there is a fix in the latest master which hopefully will be included in the next stable release :)

I saw [#148333](https://github.com/flutter/flutter/issues/148333) but I think that's a different issue - according to the comments that was fixed in 3.22.1, and this issue seems to still be present in 3.22.1

**Comment by darshankawar** [comment]
Closing this as fixed.
